pipelines/templates/build-golang-template.yaml

```
@@ -0,0 +1,54 @@
+# This template builds a golang container and pushes it to a container registry, using presidio Makefile.
+# It is used in CI to build the base container as well as all golang based services.
+# parameters:
+#   registry_service_connection: service connection to docker registry.
+#   registry_name: full name of container registry (i.e. presidio.azurecr.io)
+#   make_command_to_run: the phony command in Makefile to execute.
+#   image_name: image name to build.
+#   presidio_label: image label.
+#   presidio_branch_label: final label the container will have (latest or branch-name)
+#   deps_label: label of deps container used when building this container.
+
+parameters:
+  registry_service_connection: '' 
+  registry_name: '' 
+  make_command_to_run: ''
+  image_name: ''
+  presidio_label: ''
+  presidio_branch_label: ''
+  deps_label: ''
+steps:
+- bash: |
+    mkdir -p '$(GOBIN)'
+    mkdir -p '$(GOPATH)/pkg'
+    rm -rf '$(MODULEPATH)'
+    mkdir -p '$(MODULEPATH)'
+    shopt -s extglob
+    mv !(gopath) '$(MODULEPATH)' --force
+    echo '##vso[task.prependpath]$(GOBIN)'
+    echo '##vso[task.prependpath]$(GOROOT)/bin'
+  displayName: 'Setup Go Env'
+- task: Docker@2
+  displayName: 'Docker Login'
+  inputs:
+    containerRegistry: ${{ parameters.registry_service_connection }}
+    command: 'login'
+- bash: |
+    make DOCKER_REGISTRY=$REGISTRY_NAME PRESIDIO_LABEL=$PRESIDIO_LABEL PRESIDIO_BRANCH_LABEL=$BRANCH_LABEL PRESIDIO_DEPS_LABEL=$PRESIDIO_DEPS_LABEL GOLANG_IMAGE=$GOLANG_IMAGE $DOCKER_COMMAND
+  env:
+    REGISTRY_NAME: ${{ parameters.registry_name }}
+    PRESIDIO_LABEL: ${{parameters.presidio_label}}
+    PRESIDIO_DEPS_LABEL: ${{parameters.deps_label}}
+    BRANCH_LABEL: ${{parameters.presidio_branch_label}}
+    DOCKER_COMMAND: ${{ parameters.make_command_to_run }}
+    GOLANG_IMAGE: ${{ parameters.image_name }}
+  workingDirectory: '$(MODULEPATH)'
+  displayName: 'Build image'
+- bash: | 
+    make DOCKER_REGISTRY=$REGISTRY_NAME PRESIDIO_LABEL=$PRESIDIO_LABEL GOLANG_IMAGE=$GOLANG_IMAGE docker-push-golang-single
```

```suggestion
    make docker-push-golang-single DOCKER_REGISTRY=$REGISTRY_NAME PRESIDIO_LABEL=$PRESIDIO_LABEL GOLANG_IMAGE=$GOLANG_IMAGE
```

why the single command and not the other one?

---

this is not done in parallel. see the pipeline: https://dev.azure.com/csedevil/Presidio/_build/results?buildId=8162&view=results